Core/include/Acts/AmbiguityResolution/AthenaAmbiguityResolution.ipp

```diff
@@ -0,0 +1,406 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/AmbiguityResolution/AthenaAmbiguityResolution.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/VectorMultiTrajectory.hpp"
+#include "Acts/EventData/VectorTrackContainer.hpp"
+#include "Acts/Utilities/VectorHelpers.hpp"
+
+#include <unordered_map>
+
+namespace Acts {
+
+template <typename track_container_t, typename traj_t,
+          template <typename> class holder_t>
+const TrackContainer<track_container_t, traj_t, holder_t>
+Acts::AthenaAmbiguityResolution::prepareOutputTrack(
+    const TrackContainer<track_container_t, traj_t, holder_t>& tracks,
+    std::vector<std::size_t>& goodTracks) const {
+  auto trackStateContainer = tracks.trackStateContainerHolder();
+  auto trackContainer = std::make_shared<VectorTrackContainer>();
```

Comment:
I don't think we should assume a backend here. We should rather make this work for all backends or move it out of Core

Reply:
This was a mistake, the whole function was not needed. The equivalent of this is there in examples. Hence I removed that function entirely